charts/consul/templates/partition-service.yaml

```diff
@@ -0,0 +1,58 @@
+{{- if (and .Values.global.adminPartitions.enabled (or (and (ne (.Values.server.enabled | toString) "-") .Values.server.enabled) (and (eq (.Values.server.enabled | toString) "-") .Values.global.enabled))) }}
+# Headless service for Consul server DNS entries. This service should only
+# point to Consul servers. For access to an agent, one should assume that
+# the agent is installed locally on the node and the NODE_IP should be used.
+# If the node can't run a Consul agent, then this service can be used to
+# communicate directly to a server agent.
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "consul.fullname" . }}-partition-service
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "consul.name" . }}
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    component: server
+  annotations:
+    {{- if .Values.global.adminPartitions.service.annotations }}
+    {{ tpl .Values.global.adminPartitions.service.annotations . | nindent 4 | trim }}
+    {{- end }}
+    # This must be set in addition to publishNotReadyAddresses due
+    # to an open issue where it may not work:
+    # https://github.com/kubernetes/kubernetes/issues/58662
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+spec:
+  type: LoadBalancer
```

Comment:
I like the consistency of always relying on the same service to "discover" Consul Servers if that makes sense. I think we should allow users to configure the type and use NodePort so that it is supported on `kind`. I have also enabled the ability to select a nodePort for https, rpc and serf

Reply:
That makes sense to me 😃